test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_session]: block_session.c
+ * The block manager extent list consists of both extent blocks and size blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_and_cleanup_ext_block(WT_EXT *ext)
+{
+    REQUIRE(ext != nullptr);
+    REQUIRE(ext->depth != 0);
+    __wt_free(nullptr, ext);
+}
+
+void
+validate_and_cleanup_size_block(WT_SIZE *size)
+{
+    REQUIRE(size != nullptr);
+    __wt_free(nullptr, size);
+}
+
+void
+cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_EXT *curr = bms->ext_cache;
+    WT_EXT *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_EXT *curr = bms->ext_cache;
+
+    REQUIRE(bms->ext_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->ext_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_ext_list(bms);
+}
+
+void
+cleanup_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    WT_SIZE *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_SIZE *curr = bms->sz_cache;
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->sz_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_EXT *ext;
+
+    ext = nullptr;
+    __wt_random_init(&session->getWtSessionImpl()->rnd);
+
+    REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    validate_and_cleanup_ext_block(ext);
```

Comment:
I think this is insufficient validation? The function does the following:

s- Callocs memory of size WT_EXT + skipdepth * 2 * pointer size. 
- Sets the ext->depth field which is whatever skipdepth was.

I can see we check that depth != 0 which makes sense but we don't check any of the other field in the memory that we now own. Technically the following should _also_ be 0:
- ext->off
- ext->size
- ext->next[0] to skipdepth should all be NULL pointers.

Now one could argue we're validating calloc but I think validating the skipdepth logic would be valuable.

Reply:
I have expanded the validation to check for this now.